main.nf

```diff
@@ -1359,15 +1359,55 @@ process samtools_flagstat_after_filter {
     file(bam) from ch_filtering_for_flagstat
 
     output:
-    file "*.stats" into ch_bam_filtered_flagstat_for_multiqc
+    tuple val(prefix), file("*stats") into ch_bam_filtered_flagstat_for_multiqc, ch_bam_filtered_flagstat_for_endorspy
 
     script:
-    prefix = "$bam" - ~/(\.bam)?$/
+    prefix = "$bam" - ~/(\.bam.filtered.bam)?$/
     """
     samtools flagstat $bam > ${prefix}_postfilterflagstat.stats
     """
 }
 
+/*
+* Step 4c: Keep unmapped/remove unmapped reads flagstat
+*/
+
+// merge tuples, for when filtered has been run
+
+//ch_bam_filtered_flagstat_for_endorspy.view {it -> "DoubleBefore: $it"}
+
+if (params.run_bam_filtering) {
+  ch_flagstat_for_endorspy
+    .join(ch_bam_filtered_flagstat_for_endorspy)
+    .set{ ch_allflagstats_for_endorspy }
+
+} else {
+  ch_flagstat_for_endorspy
+    .groupTuple(by: 0)
+    .set{ ch_allflagstats_for_endorspy }
+}
+
+process endorSpy {
+    label 'sc_tiny'
+    tag "$prefix"
+    publishDir "${params.outdir}/endorSpy", mode: 'copy'
+
+    when:
+    !params.skip_mapping
+
+    input:
+    tuple val(name), file(stats), file(poststats) from ch_allflagstats_for_endorspy
+
+    output:
+    file "*.json" into ch_endorspy_for_multiqc
+
+    script:
+    prefix = "${name}"
+    """
+    endorS.py ${stats} ${poststats} -o json -n ${name}
```

Comment:
You will need to revise this line after modifying the argument parsing (files go after flags)

Reply:
Could you re-shuffle the order of the options (first), and arguments (last) @aidaanva ?

Reply:
@maxibor the options and the arguments can be reshuffled, @jfy133 should update the code now